### PR TITLE
Support updating username when matching password provided

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203058117400612/f

### Description
Detects when a saved login is missing a username, and offers to save the username when logging in again (when a matching password is included)

#### Updating usernames for same site
- [x] Save 3 logins for the same site, **using the same password for each** (different usernames)
- [x] Edit one of the logins: remove the username
- [x] Save a few other logins for the same site using different usernames and passwords
- [x] Refresh the login page and refuse any offers to autofill
- [x] Login using a unique username with the same password from step 1
- [x] Verify it offers you to update your username, but choose not to
- [x] Login using a unique username with the same password from step 1
- [x] This time, accept its offer to update your username
- [x] Visit autofill logins in settings and verify the username was updated for only the entry which matched this password but was missing the username

#### No updating usernames for a different site
- [x] Clear all autofill entries
- [x] Add a saved login for a site and remove its username manually
- [x] Visit another site and attempt a login using the same password from above
- [x] Verify you are not offered to update the username

#### Updating password
- [x] Clear all autofill entries
- [x] Add a saved login for a site
- [x] Revisit login page and attempt a login using the same username as before. 
- [x] Verify you are prompted to update password. Say not now, and verify the password wasn't updated
- [x] Revisit login page and attempt a login using the same username as before. 
- [x] Verify you are prompted to update password. Agree, and verify the password was updated correctly

### CODE REVIEW

**Comment by CDRussell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2434** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2434" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2434?utm_source=stack-comment).

**Review by PIYUSH-MISHRA-00 — COMMENTED** [review]
implemented

**Comment by CDRussell** [comment]
Translations will come in https://github.com/duckduckgo/Android/pull/2460

File: app/src/main/java/com/duckduckgo/app/browser/autofill/AutofillCredentialsSelectionResultHandler.kt [hunk 1]
```diff
@@ -152,7 +153,9 @@ class AutofillCredentialsSelectionResultHandler @Inject constructor(
     ): LoginCredentials? {
         val selectedCredentials = result.getParcelable<LoginCredentials>(CredentialUpdateExistingCredentialsDialog.KEY_CREDENTIALS) ?: return null
         val originalUrl = result.getString(CredentialUpdateExistingCredentialsDialog.KEY_URL) ?: return null
-        return credentialSaver.updateCredentials(originalUrl, selectedCredentials)
+        val updateType =
```
Comment (karlenDimla) [inline app/src/main/java/com/duckduckgo/app/browser/autofill/AutofillCredentialsSelectionResultHandler.kt:156]: updateType can be null here it seems but AutofillCredentialSaver.updateCredentials() doesn't accept a null CredentialUpdateType. (https://github.com/duckduckgo/Android/pull/2434/files#diff-5464b8c097bf98dc5e4bd9bd3108916a2a9800e28630d2b1d1a38480ed2be433R185)

Comment (karlenDimla) [reply app/src/main/java/com/duckduckgo/app/browser/autofill/AutofillCredentialsSelectionResultHandler.kt:156]: I guess same goes for originalUrl and selectedCredentials.

**Review by karlenDimla — APPROVED** [review]

**Pull request merged**